tools/building.py

```diff
@@ -1339,7 +1339,6 @@ def is_wasm_dylib(filename):
 def map_to_js_libs(library_name):
   # Some native libraries are implemented in Emscripten as system side JS libraries
   library_map = {
-    'c': [],
```

Comment:
why is it safe to remove this?

Reply:
Its not just safe, but essential if we want to continue to allow folks to do `-lc` on the command line.

The reason is that previously we would search for libc.a on the filesystem before looking in this map.  With this change we call map_to_js_libs first, so this would cause `-lc` to be ignored, which we don't want.

In practice anyone who passes `-lc` today might not get the correct version of libc, but that was true both before and after this change.   To fix that issue we have #14355, although that issue is completely separate to this change.

Reply:
I am having a hard time understanding this. How is, say, `libc` (which is removed here) different from `libdl` (which is not)?

Reply:
`libdl` and `libm` don't exist under musl, since they all get rolled into `libc`.   So the `libdl` and `libm` cases here are different because we really do want them to be no-ops.

Reply:
We are pretending to have `-ldl` support for compat with glibc

Reply:
In that case, how about comments on each one explaining the issues? Sorry, but this is all quite unclear to me. Maybe this is just part of the codebase I am unfamiliar with.

Reply:
I agree..  that would be more clear.

Reply:
After rebasing on top of #14382 perhaps this is more clear now?